flask_gae_blobstore.py

```diff
@@ -274,40 +281,40 @@ def validate_file_type(result, accept_file_types=UPLOAD_ACCEPT_FILE_TYPES):
     return True
 
 
-def write_to_blobstore(data, mime_type, name=None):
-    '''Writes a file to the App Engine blobstore and returns an instance of a
-    BlobKey if successful.
+def write_to_gcs(data, mime_type, name=None):
+    '''Writes a file to Google Cloud Storage and returns the file name
+    if successful.
 
-      :param data: Blob data.
-      :param mime_type: String, mime type of the blob.
-      :param name: String, name of the blob.
+      :param data: Data to be stored.
+      :param mime_type: String, mime type of the data.
+      :param name: String, name of the data.
 
-      :returns: Instance of a `BlobKey`.
+      :returns: String, filename.
     '''
     if not name:
         name = ''.join(random.choice(string.letters)
                        for x in range(DEFAULT_NAME_LEN))
 
-    blob = files.blobstore.create(
-        mime_type=mime_type,
-        _blobinfo_uploaded_filename=name)
-    with files.open(blob, 'a', exclusive_lock=True) as f:
-        f.write(data)
-    files.finalize(blob)
-    result = files.blobstore.get_blob_key(blob)
-    # issue with the local development SDK. we can only write to the blobstore
-    # so fast, so set a retry_count and delay the execution thread between
-    # each attempt..
-    for i in range(1, WRITE_MAX_RETRIES):
-        if result:
-            break
-        else:
-            logging.debug(
-                'blob still None.. will retry to write to blobstore..')
-            time.sleep(WRITE_SLEEP_SECONDS)
-            result = files.blobstore.get_blob_key(blob)
-        logging.debug('File written to blobstore: key: "%s"', result)
-    return result
+    bucket_name = app_identity.get_default_gcs_bucket_name()
+
+    resume_uuid = uuid.uuid4()
+    bucket_filename = '/' + bucket_name + '/' + str(resume_uuid)
+
+    write_retry_params = gcs.RetryParams(backoff_factor=1.1)
+    gcs_file = gcs.open(bucket_filename,
+                        'w',
+                        content_type=mime_type,
+                        options = {
+                            'x-goog-meta-filename': name,
+                            'Content-Disposition': 'attachment; filename={}'
+                                .format(name)
+                        },
```

Comment:
Content-disposition, write_retry_params & bucket_name seem like parameters that we might want to change without having to change this library.